### DISCUSSION TITLE
Use Rest API as Data Source

### DISCUSSION THREAD

**Discussion by SaschaValentin** [mavoweb/mavo#965] [Q&A]
In a former post I discussed if there are connections connections in Mavo to realtiome backends. Even so this is not possible yet, I thought about a kind of workaround.

So I wonder if it is possible to fetch data from a Rest API, write it as an Object to a variable and use this as a datasource for a Mavo app.

**Comment by LeaVerou** [comment]
If the API doesn't need authentication, you can already do this! Any URL works in `mv-storage`, `mv-source`, `mv-init`!
If it does require authentication, it would require a custom backend to define a few things about *how* authentication works in that API, which would require some light programming.

**Reply by SaschaValentin** [reply]
Hello Lea,

thanks for your response.

Because of the authentication I first wanted to write the result into an variable. But I don´t know if Mavo accepts a variable as datasource.

**Reply by karger** [reply]
Note that Tarfah Alrashed implemented, for her PhD project, a general infrastructure for connecting Mavo as front end to a wide variety of API back ends including authentication.   https://shapir.org/mavo-shapir

**Comment by Akshat111111** [comment]
It is vital to provide the required authentication headers in your HTTP requests when interacting with a REST API that requires authentication. This is how you retrieve data from a verified REST API, save it in an object, and utilize it as a source of data for a Mavo application.Establish Authentication,Store data as an object, get data from an authenticated REST API and finally Combine with Mavo
